### DISCUSSION TITLE
Templater Plugin Syntax Not Working

### DISCUSSION THREAD

**Discussion by margarita-rivera** [SilentVoid13/Templater#766] [Help]
I've presented this in the [Bryan Jenks Community](https://github.com/BryanJenksCommunity/FAQ/discussions/146) - work flow seems to be okay so he suggested to reach out here if in case it might be a technical issue

### **Issue:** 
Templater syntax not working

### **Example:**

<%tp.date.now()%> remains <%tp.date.now()%> instead of:

<%tp.date.now()%> to 2022-04-23

I've uninstalled/installed templates I am using and in loading a template it still shows this way:

<img width="1244" alt="Screen Shot 2022-04-23 at 7 28 31 PM" src="https://user-images.githubusercontent.com/104243828/164892719-731ebeaa-44a9-46b3-a5e6-f61bf3a9c947.png">

### **Settings/Other actions taken:**

- Installed and enabled templater plugin, made sure it is updated (prompted by [discussion](https://github.com/BryanJenksCommunity/FAQ/discussions/78))

- Set templater folder to my “Templates” folder and my settings as:

<img width="997" alt="Screen Shot 2022-04-23 at 7 29 40 PM" src="https://user-images.githubusercontent.com/104243828/164892743-6af96c86-2e98-4bba-808a-04c8c72f734a.png">

<img width="986" alt="Screen Shot 2022-04-23 at 7 29 48 PM" src="https://user-images.githubusercontent.com/104243828/164892746-8725f499-240b-456f-aca4-30bacbb35a7a.png">

- Turned off core plugin “templates” (prompted by [discussion](https://github.com/BryanJenksCommunity/FAQ/discussions/60)) 

<img width="1622" alt="Screen Shot 2022-04-24 at 11 50 41 AM" src="https://user-images.githubusercontent.com/104243828/164955806-6c8f80c4-f0da-40f9-a6ea-b24c7a66d63e.png">

- customized hotkeys of templater (prompted by [discussion](https://github.com/BryanJenksCommunity/FAQ/discussions/60))

<img width="996" alt="Screen Shot 2022-04-23 at 7 30 39 PM" src="https://user-images.githubusercontent.com/104243828/164892727-fc39b0be-9830-415c-b33c-bf89ff95ec4d.png">

Thanks for much taking time to read this! Hoping I just missed some setting here or something and this works soon *fingers crossed*

**Comment by AB1908** [comment]
Does replacing templates in file not resolve those commands?

**Comment by woocore** [comment]
I have the same problems, too.

**Comment by welpdx** [comment]
Would you be able create a new vault with the bare minimum set up and see if the issue persists?
If it does, can you upload that vault here?

I ask because I'm sorry but couldn't reproduce this issue
<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/Y8gCo4R.gif" alt="demo" />
</details>

---
Edit: hid demo gif

**Comment by ghost** [comment]
I seem to be having this issue as well. None of the docs snippets render. I have tried:
- creating a new vault with nothing but templater.
- removed and reinstalled templater.
- tried both AppImage and flatpak releases of Obsidian. 

None of these have resolved the problem.

**Comment by welpdx** [comment]
@Miusaky, How are you executing Obsidian commands? (I.e., `<% tp.date.now() %>`)

### There are 5 ways to execute Obsidian Comands. 

#### You can execute Obsidian Commands that start with `<%`  and `<%*` via these 3 ways:

1. **Create new note from template. (Default hotkey: `Alt + N`)** 
   <details>
           <summary style="font-size:12">new note from template</summary>
           <img src="https://i.imgur.com/7wGFaIS.gif" alt="demo" />
   </details>
2. **Open Insert Template modal.  (Default hotkey: `Alt + E`)** 
   <details>
           <summary style="font-size:12">new note from template</summary>
           <img src="https://i.imgur.com/dZMoKYe.gif" alt="demo" />
   </details>
3. **Replace templates in the active file. (Default hotkey: `Alt + R`)** 
   <details>
           <summary style="font-size:12">Replace templates in the active file </summary>
           <img src="https://i.imgur.com/JtRLYkW.gif" alt="demo" />
   </details>

#### You can also use [dynamic commands `<%+`](https://silentvoid13.github.io/Templater/commands/dynamic-command.html):

4. **Using Dynamic Commands by switching from writing to reading mode.**
   <details>
           <summary style="font-size:12">Demo with Dynamic Commands</summary>
           <img src="https://i.imgur.com/xo8G0Xt.gif" alt="demo" />
   </details>


5. **Using Folder Templates. Adds Template content in new file, and executing Templater Commands**
   <details>
           <summary style="font-size:12">Using Folder Template</summary>
           <img src="https://i.imgur.com/HX8fHZ7.gif" alt="demo" />
   </details>

**Comment by ghost** [comment]
@welpdx I needed to enable 'Trigger Templater on new file creation'. It renders the date now. Should have read through the settings first. Thanks.

**Comment by welpdx** [comment]
@Miusaky Thanks for reporting back! I'm happy that you were able to fix your issue.

I had thought you had the same settings as @margarita-rivera, and that is why I did not mention the `Trigger Templater on new file creation` setting. Sorry.

By the way, I also didn't read the settings closely when I first started using it lol. It took me a while to get Templater working because I kept thinking that switching to obsidian's `reading mode` could execute static Templater commands (`<%` `<%*`) ! 🤦‍♂️
